index.html

```diff
@@ -1917,28 +1917,29 @@
                 If <var>record</var>.recordType is <code>undefined</code>, then:
                 <ol> <!-- guess type and mediaType from data -->
                   <li>
-                    If <var>record</var>.data is instance of
-                    <code>ArrayBuffer</code>, then
-                    set <var>record</var>.mediaType to
-                    <code>"application/octet-stream"</code> and
-                    <var>record</var>.recordType to <code>"opaque"</code>.
+                    If the type of <var>record</var>.data is an
+                    <code>ArrayBuffer</code>, set <var>record</var>.recordType
+                    to <code>"opaque"</code> and if <var>record</var>.mediaType
+                    is an empty string, set <var>record</var>.mediaType to
+                    <code>"application/octet-stream"</code>.
                   </li>
                   <li>
-                    Otherwise, if the type of <var>record</var>.data is
-                    <code>"object"</code>, then set
-                    <var>record</var>.mediaType to
-                    <code>"application/json"</code> and
-                    <var>record</var>.recordType to <code>"json"</code>.
+                    Otherwise, if the type of <var>record</var>.data is an
+                    <code>Object</code>, set <var>record</var>.recordType
+                    to <code>"json"</code> and if <var>record</var>.mediaType
+                    is an empty string, set <var>record</var>.mediaType to
+                    <code>"application/json"</code>.
                   </li>
                   <li>
                     Otherwise, if the type of <var>record</var>.data is
-                    <code>"number"</code> or <code>"string"</code>, then
-                    set <var>record</var>.mediaType to <code>"plain/text"</code>
-                    and <var>record</var>.recordType to <code>"text"</code>.
+                    <code>UnrestrictedDouble</code> or <code>String</code>, then
+                    set <var>record</var>.recordType to <code>"text"</code>
+                    and if <var>record</var>.mediaType is an empty string,
```

Comment:
I tried that, was looking funny, every step had just 1 substep. Do you want me to upload version with substeps?